lib/msf/ui/console/table_print/custom_color_styler.rb

```diff
@@ -0,0 +1,39 @@
+# -*- coding: binary -*-
+
+module Msf
+  module Ui
+    module Console
+      module TablePrint
+        class CustomColorStyler
+          RESET_COLOR = '%clr'.freeze
+          # @param [Hash<String, String>] opts A hash of a String to find, and what color to use
+          # For example: opts = { 'abc' => '%grn' }
+          def initialize(opts)
+            @highlight_terms = {}
+            opts.each { |term, color| @highlight_terms[term] = color }
+          end
+
+          def style(value)
+            if @highlight_terms.key?(value)
+              return "#{@highlight_terms[value]}#{value}#{RESET_COLOR}"
+            end
+
+            colored_value = value
+            @highlight_terms.each_key do |key|
+              if value.include?(key)
+                colored_value.gsub!(key, "#{@highlight_terms[key]}#{key}#{RESET_COLOR}")
```

Comment:
I think this implementation suffers the same fate as the previous highlighter for the search table  https://github.com/rapid7/metasploit-framework/pull/14930